research/conditional/moe_layers/token_choice.py

```diff
@@ -63,14 +65,18 @@ def forward(self, x: torch.Tensor):
         capacity = int(
             self.capacity_factor * n_tokens * self.routing_top_k / self.n_experts
         )
+        if self.vectorize:
+            capacity = min(capacity, n_tokens)
```

Comment:
This is because vectorize doesn't work with routing_top_k > 1?

Reply:
With vectorize we use topk and k in topk cannot be greater than the dimension length